src/integration.test.ts

```diff
@@ -32,6 +39,34 @@ describe("Integration Test", (): void => {
 
     inMemoryCache = {};
     state = {};
+    loadCommand = `docker load --input ${docker.DOCKER_IMAGES_PATH}`;
+
+    let callCount = 0;
+
+    child_process.exec.mockImplementation(
+      (_command: any, _options: any, callback: any): any => {
+        let stdout;
+        if (
+          _command ===
+          'docker image list --format "{{ .Repository }}:{{ .Tag }}"'
+        ) {
+          /* When Docker is running as root, docker image list generates a list that includes a
+           * standard set of Docker images that are pre-cached by GitHub Actions. The production
+           * code filters out the Docker images present during the restore step from the list of
+           * images to save since caching pre-cached images would harm performance and waste
+           * cache space. This mock implementation of docker image list ensures a non-empty
+           * difference between the restore and save steps so there is something to save.
+           */
+          callCount += 1;
+          dockerImages.push(`test-docker-image:v${callCount}`);
+          stdout = dockerImages.join("\n");
+        } else {
+          stdout = STDOUT;
+        }
+        callback(null, { stdout, stderr: STDERR });
+        expect(core.info).lastCalledWith(stdout);
```

Comment:
This is racy; there's no guarantee that it executes after `util.execAsPromised` returns. The race condition can be eliminated by moving this line to `assertExecBashCommand`.